### PR TITLE
Bugfix/Feature: pascalCase; Bugfix: recursive nesting

### PR DESCRIPTION
**Pull request by bohdanbirdie**
This PR is going to fix the issue with pascalCase error, initially committed by @keesvanlierop
Also, I extended that functionality to support upper case enums as well.

The other thing that was failing for me was a recursive nesting.
Check this schema
```graphql
type User {
  id: Int!
  username: String!
  tasks: [Task!]!
}

type Task {
  id: Int!
  description: String
  author: User!
  status: TaskStatus!
}
```

so, when I create a mock of a task -> user mock will be created, but then user mock will create a task mock and this will run recursively until the stack can't hold it anymore.
Overrides won't help because they come **after** the mock is created so I switched to the getters, so they won't create a nested mock if there is an override.

### CODE REVIEW

**Comment by bohdanbirdie** [comment]
ok, @ardeois should be good for the review 👍

File: package.json [hunk 1]
```diff
@@ -43,7 +44,7 @@
         "eslint-plugin-import": "^2.17.2",
         "eslint-plugin-jest": "^22.20.0",
         "eslint-plugin-prettier": "^3.0.1",
-        "graphql": "^14.5.8",
+        "graphql": "^14.6.0",
```
[the remark this answers is not in the document]
Comment (bohdanbirdie) [reply package.json:47]: Would you like to to rever it?
I think I had issues with different versions of graphql so had to update

[the remark this answers is not in the document]
Comment (ardeois) [reply package.json:47]: I just merged the update dependencies PR, so if you update your branch you should be good

**Comment by ardeois** [comment]
@bohdanbirdie You should be good to update your branch with master.
I've updated all dependencies and added precommit hook that should run prettier for you. So it will format the `README.md` properly

**Comment by bohdanbirdie** [comment]
@ardeois ok, updated everything, synced, added test and changed to pascal case as thee default

**Pull request merged**